src/components/drawing/index.js

```diff
@@ -500,7 +529,8 @@ drawing.pattern = function(sel, gd, patternID, shape, bgcolor, fgcolor, size, so
                 'id': fullID,
                 'width': width + 'px',
                 'height': height + 'px',
-                'patternUnits': 'userSpaceOnUse'
+                'patternUnits': 'userSpaceOnUse',
+                'patternTransform': isLegend ? 'scale(0.7)' : ''
```

Comment:
Can we adjust the size from the legend side, rather than detecting the legend inside here? Also do we need to scale down *all* these sizes? I'd rather we just restrict the maximum size to something we know will work, and perhaps when we have an array for pattern size just use the default size instead of the first one.

Reply:
Good calls.
See https://github.com/plotly/plotly.js/pull/5733/commits/5f5561f0269931b59081c665691696b953123a48, https://github.com/plotly/plotly.js/pull/5733/commits/a2a66e7d539859655db759361b79d79dea16fd7c and https://github.com/plotly/plotly.js/pull/5733/commits/69f91e2c3921667d5e6f5f434eddde842a71f49c.

Reply:
Again though, do we need to scale down to 0.7 here? In the [legend style we already limit the size to 12](https://github.com/plotly/plotly.js/pull/5733/files#diff-3f7cb083e91b0a4c750484d81114195ba135277cb648d2b6751cdeaeacaffacfR369), can't we keep the same scale as on the graph, but if 12 is too big just reduce that limit?

Reply:
Today in https://github.com/plotly/plotly.js/pull/5733/commits/38bd5ec0ea4a5ef7b6d9edbca822d64ca7bace81 I did try to avoid this scale reduction. But the problem is that the default size of 8 does not fit nicely in the legend icons and lose their identity.
So again in https://github.com/plotly/plotly.js/pull/5733/commits/178e0a608a2ffa37f111f8c86e97a042038eba37 I scaled down the patterns for icons using a greater ratio at 0.8.

Reply:
I think the current legend items are "ok" ... better than on master, and we can always tweak them a bit more later. I really do need to get this merged and released so unless there's a hard blocker/pathological case here, can we defer the remaining work @alexcjohnson ?

Reply:
OK